core/src/main/java/com/microsoft/applicationinsights/internal/config/ReflectionUtils.java

```diff
@@ -56,9 +88,14 @@
                 return null;
             }
 
-            Class<?> clazz = Class.forName(className).asSubclass(interfaceClass);
+            Class<?> clazz = builtInMap.get(className);
+            if (clazz == null) {
+                InternalLogger.INSTANCE.trace(" :( class lookup failed: %s", className);
+                clazz = Class.forName(className).asSubclass(interfaceClass);
```

Comment:
Please add a comment to mention this as a fallback when class is not loaded. It would be difficult to understand later on.

Reply:
I think the conditional itself is pretty clear. What would you have the comment say?